File: Day2/c++/src/main.cpp
#include "functions.h"
#include <fstream>
#include <iostream>
#include <string>

int main()
{
	int sum =0;
	int totalPower=0;
	std::ifstream file("input.txt");

	if(!file.is_open()){
		std::cerr<<"Error opening file";
		return 1;
	}

	std::string line;

	while (std::getline(file, line)){
		line = line.substr(0,line.length()-1);
		Game game(line);
		if(game.isPossible())
			sum += game.gameNumber();
		totalPower += game.getPower();
	}

	file.close();
	std::cout<<"Total Sum of Possible Games:"<<std::endl<<sum<<std::endl;
	std::cout<<"Total Power"<<std::endl<<totalPower<<std::endl;
	return 0;
}

File: Day2/go/main.go
package main

import (
	"fmt"
	"bufio"
	"os"
)

func main(){
	sum := 0
	power := 0
	//open file
	file, err := os.Open("input.txt")
	if err !=nil {
		fmt.Fprintln(os.Stderr, "Error opening file:", err)
		os.Exit(1)
	}
	
	//Close file after use. 
	defer file.Close()

	//read line from file
	scanner := bufio.NewScanner(file)
	for scanner.Scan(){
		line:=scanner.Text()
		game := parseGame(line)
		if game.isPossible{
			sum += game.gameNumber
		}
		power += (game.redMax * game.greenMax * game.blueMax)
		
		
	}

	//output result

		fmt.Println(sum)
		fmt.Println(power)


}
